support/releaseinfo/registry_image_content_policies.go

```diff
@@ -32,8 +34,12 @@ func (p *ProviderWithOpenShiftImageRegistryOverridesDecorator) Lookup(ctx contex
 				// Attempt to lookup image with mirror registry destination
 				releaseImage, err := p.Delegate.Lookup(ctx, image, pullSecret)
 				if releaseImage != nil {
-					p.mirroredReleaseImage = image
-					return releaseImage, nil
+					// Verify mirror image availability.
+					if _, _, err = registryclient.GetRepoSetup(ctx, image, pullSecret); err == nil {
+						p.mirroredReleaseImage = image
+						return releaseImage, nil
+					}
+					continue
 				}
 
 				logger.Error(err, "Failed to look up release image using registry mirror", "registry mirror", registryReplacement)
```

Comment:
Should this be a warning instead since we are potentially looking through lots of mirrors?